frb_codegen/src/main.rs

```diff
@@ -26,3 +26,195 @@ fn main() -> anyhow::Result<()> {
+        };
+        use std::process::Command;
+
+        set_dir();
+
+        /// Path of input Rust code
+        const RUST_INPUT: &str = "../frb_example/pure_dart/rust/src/api.rs";
+        /// Path of output generated Dart code
+        const DART_OUTPUT: &str = "../frb_example/pure_dart//dart/lib/bridge_generated.dart";
+
+        let _ = *LOGGER;
+
+        // Tell Cargo that if the input Rust code changes, to rerun this build script.
+        println!("cargo:rerun-if-changed={RUST_INPUT}");
+        // Options for frb_codegen
+        let raw_opts = RawOpts {
+            // Path of input Rust code
+            rust_input: vec![RUST_INPUT.to_string()],
+            // Path of output generated Dart code
+            dart_output: vec![DART_OUTPUT.to_string()],
+            wasm: true,
+            dart_decl_output: Some(
+                "../frb_example/pure_dart/dart/lib/bridge_definitions.dart".into(),
+            ),
+            dart_format_line_length: 120,
+            // (extra) c output path
+            c_output: Some(vec![
+                // each field should contain head file name
+                "../frb_example/pure_dart/rust/c_output_path/c_output.h".into(),
+            ]),
+            extra_c_output_path: Some(vec![
+                "../frb_example/pure_dart/rust/c_output_path_extra/".into()
+            ]),
+
+            // for other options use defaults
+            ..Default::default()
+        };
+
+        // get opts from raw opts
+        let all_configs = config_parse(raw_opts);
+
+        // generation of rust api for ffi (single block)
+        let all_symbols = get_symbols_if_no_duplicates(&all_configs).unwrap();
+        assert_eq!(all_configs.len(), 1);
+        frb_codegen(&all_configs[0], &all_symbols).unwrap();
+
+        let status = Command::new("cargo")
+            .current_dir("../frb_example/pure_dart/rust")
+            .arg("build")
+            .spawn()
+            .expect("failed to execute cargo")
+            .wait()
+            .expect("failed to wait for cargo to finish");
+        assert!(status.success(), "cargo build failed");
+
+        let status = Command::new("dart")
+            .arg("../frb_example/pure_dart/dart/lib/main.dart")
+            .arg("../target/debug/libflutter_rust_bridge_example.so")
+            .spawn()
+            .expect("failed to execute pure_dart")
+            .wait()
+            .expect("failed to wait for pure_dart");
+        assert!(status.success(), "pure_dart failed");
+    }
+
+    #[test]
+    fn pure_dart_multi() {
+        // `cargo build` can fail because frb_example/pure_dart makes testing a build step! *gasp*
+        // If you want to get `cargo build` working again and investigate why pure_dart fails:
+        // 1. mv ../frb_example/pure_dart_multi/rust/build.rs ../rb_example/pure_dart_multi/rust/_build.rs
+        // 2. Run this test in the debugger - which is a copy of build.rs (but more complete)
```

Comment:
uber-nit: avoid copying comments, but just say "see comments of the pure_dart test above"

Reply:
fixed